### PR TITLE
mouse spec

### PR DESCRIPTION
**Pull request by pelikhan**
Support for HID Mouse drivers.

### CODE REVIEW

File: services/mouse.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
+# Mouse
+
+    identifier: 0x1885dc1c
+
+Control a device that acts as a mouse.
+
+## Commands
+
+    enum Button : u8 {
```
Comment (mmoskal) [inline services/mouse.md:9]:
```suggestion
    flags Button : u8 {
```

File: services/mouse.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+        right = 0x01
+        middle = 0x04
+        left = 0x02
+    }
+    enum ButtonEvent : u8 {
+        up = 0x01
+        down = 0x02
+        click = 0x03
+        double_click = 0x04
```
Comment (mmoskal) [inline services/mouse.md:18]: I think we use `DoubleClick` casing elsewhere

File: services/mouse.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    }
+    command set @ 0x80 {
+        buttons: Button
+        event: ButtonEvent
+    }
+
+Sets the up/down state of one or more buttons.
+
+    command move @ 0x81 {
+      x: i8
```
Comment (mmoskal) [inline services/mouse.md:28]: dx/dy?

File: services/mouse.md [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+      y: i8
```
Comment (mmoskal) [inline services/mouse.md:29]: what is the unit of the distance? (could be in comment)
also maybe use i16

File: services/mouse.md [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+    }
+
+
+Moves the mouse by the distance specified.
+
+    command wheel: i8 @ 0x82
+
+Turns the wheel up or down.
```
Comment (mmoskal) [inline services/mouse.md:37]: would be good to say "positive is scroll up" (if mouse set to "natural scrolling") or something

File: services/mouse.md (hunk 3 above)
Comment (mmoskal) [reply services/mouse.md:28]: also should there be a time parameter? to spread the movement over time?

File: services/mouse.md (hunk 2 above)
Comment (mmoskal) [reply services/mouse.md:18]: I don't think USB mouse is generating click or double click events, is it?

Comment (pelikhan) [reply services/mouse.md:18]: I am thinking that the device generates those by sending up/down events with the correct timing. Relying on the jacdac bus packets is probably too brittle. Sending a click is very common.

File: services/mouse.md (hunk 3 above)
Comment (pelikhan) [reply services/mouse.md:28]: yeah that's good

File: services/mouse.md (hunk 4 above)
Comment (pelikhan) [reply services/mouse.md]: done

File: services/mouse.md (hunk 2 above)
Comment (mmoskal) [reply services/mouse.md]: sounds good!

File: services/mouse.md [hunk 6]
```diff
@@ -0,0 +1,42 @@
+# Mouse
+
+    identifier: 0x1885dc1c
+
+Control a device that acts as a mouse.
+
+## Commands
+
+    flags Button : u8 {
```
Comment (mmoskal) [inline services/mouse.md:9]: I think there might be mice with more than 8 buttons
```suggestion
    flags Button : u16 {
```

File: services/mouse.md [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+        Down = 0x02
+        Click = 0x03
+        DoubleClick = 0x04
+    }
+    command set_button @ 0x80 {
+        buttons: Button
+        event: ButtonEvent
+    }
+
+Sets the up/down state of one or more buttons.
```
Comment (mmoskal) [inline services/mouse.md:25]: We should specify:

A `Click` is the same as `Down` followed by `Up` after 100ms.
A `DoubleClick` is two clicks with 150ms gap between them (that is, 100ms first click, 150ms gap, 100ms second click).

File: spectool/jdspec.ts [hunk 8]
```diff
@@ -684,6 +684,10 @@ export function parseSpecificationMarkdownToJSON(filecontent: string, includes?:
             case "high":
                 info.highCommands = !!parseIntCheck(words[2])
                 break
+            case "status":
+                if (["stable", "experimental", "deprecated"].indexOf(words[2]) > -1)
+                    info.status = <any>words[2];
```
Comment (mmoskal) [inline spectool/jdspec.ts:689]: else error("unknown status")

**Comment by pelikhan** [comment]
Commit this as is to test experimental rendering.

**Pull request merged**